pkg/monitor/cluster/healthz.go

```diff
@@ -5,17 +5,53 @@ package cluster
 
 import (
 	"strconv"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/errors"
 )
 
 func (mon *Monitor) emitAPIServerHealthzCode() (int, error) {
 	var statusCode int
-	err := mon.cli.Discovery().RESTClient().
+	raw, err := mon.cli.Discovery().RESTClient().
 		Get().
-		AbsPath("/healthz").
+		AbsPath("/readyz").
 		Do().
 		StatusCode(&statusCode).
-		Error()
+		Raw()
+
+	// HTTP 500s are expected behaviour (e.g. on startup/shutdown of the API server) and should not cause this function to return an error. Therefore, silence it.
+	if err != nil && errors.IsUnexpectedServerError(err) {
+		err = nil
+	}
```

Comment:
(also relevant to https://github.com/Azure/ARO-RP/pull/855#discussion_r447921912 from @ehashman )

So, my thinking here is that for this specific function, a 500 from the server is actually not a problem with the function, since health/readyz and shouldn't be reported with an error. A 502 (gateway issue) or 504 (gateway timeout) would still report themselves as errors. That is, I guess I'm trying to distinguish between "something happened, preventing this check from running", and "the check performed successfully, and the target is down", since the latter feels like noise that might distract from the entirely normal situation where an API server may be shutting down and readyz returns "not ready" on metrics dashboards.

That is, you can determine a likely API server state with the following signals:

| readyz | clustererrors | result |
| --- | --- | --- |
| ready = 1 | clustererrors = 0 | OK |
| ready = 1 | clustererrors >= 1 | server is ready but checks are failing |
| ready = 0 | clustererrors = 0 | server is not ready (startup, shutdown) |
| ready = 0 | clustererrors >= 1 | server is uncontactable, assume unready |

Whether this is actually useful outside of my mind is another thing, but it feels like something I would use to make alerting dashboards more useful.